gms/gossiper.cc

```diff
@@ -409,6 +409,9 @@ future<> gossiper::handle_ack2_msg(gossip_digest_ack2 msg) {
 }
 
 future<> gossiper::handle_echo_msg() {
+    if (_initial_generation_number.has_value()) {
```

Comment:
The name `_initial_generation_number` doesn't make much sense in the context of this minimal patch since we don't do anything with the generation_number value.  We just use the presence of a value to flag this function to throw.

I suggest instead to have a boolean `_advertise_myself` flag a a corresponding method to wrap the logic around it (so the low-level logic can seamlessly change in the future).
The flag can be initialized to `false` and set to `true` in `start_gossiping` if called with `gms::advertise_myself::yes` and in `gossiper::advertise_myself()`

Reply:
You know I will use _initial_generation_number very soon in the follow up patch. But I do not mind changing it to a bool flag and change it back later.